src/server/game-server.ts

```diff
@@ -37,6 +92,19 @@ export class GameServer {
     this.tick();
   }
 
+  onPregameDisconnection(socket: SocketIO.Socket) {
+    this.sockets.splice(this.sockets.indexOf(socket), 1);
+    this.connectedPlayers -= 1;
+  }
+
+  onDisconnection(playerId: string) {
```

Comment:
Could this be `onIngameDisconnection`?  `onDisconnection` implies it happens even in pregame